spring-integration-event/src/test/java/org/springframework/integration/event/dsl/IntegrationFlowEventsTests.java

```diff
@@ -176,6 +190,17 @@ public IntegrationFlow delayFlow() {
 					.channel(MessageChannels.queue("delayedResults"));
 		}
 
+		@Bean
+		QueueChannel eventFromPublisher() {
+			return new QueueChannel();
+		}
+
+		@EventListener
+		@Publisher("eventFromPublisher")
```

Comment:
While this would work, do we really want to promote it? An `ApplicationEventListeningMessageProducer` is a much simpler approach.